static/panes/opt-view.ts

```diff
@@ -0,0 +1,183 @@
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import _ from 'underscore';
+import * as monaco from 'monaco-editor';
+import { Container } from 'golden-layout';
+
+import { MonacoPane } from './pane';
+import { OptState } from './opt-view.interfaces';
+import { MonacoPaneState } from './pane.interfaces';
+
+import { ga } from '../analytics';
+import { extendConfig } from '../monaco-config';
+
+type SourceLocation = {
+    File: string;
+    Line: number;
+    Column: number;
+};
+
+type OptCodeEntry = { // TODO: Not fully correct type yet, will do for now
+    DebugLoc: SourceLocation;
+    Function: string;
+    Pass: string;
+    Name: string;
+    text: string;
+    optType: string;
+    displayString: string;
+};
+
+export class Opt extends MonacoPane<monaco.editor.IStandaloneCodeEditor, OptState> {
+    decorations: any = {};
+    currentDecorations: string[] = [];
+    // Note: bool | undef here instead of just bool because of an issue with field initialization order
+    isCompilerSupported?: boolean;
+
+    constructor(hub: any, container: Container, state: OptState & MonacoPaneState) {
+        super(hub, container, state);
+        if (state && state.optOutput) {
+            this.showOptResults(state.optOutput);
+        }
+        this.eventHub.emit('optViewOpened', this.compilerInfo.compilerId);
+    }
+
+    override getInitialHTML(): string {
+        return $('#opt').html();
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(editorRoot, extendConfig({
+            language: 'plaintext',
+            readOnly: true,
+            glyphMargin: true,
+        }));
+    }
+
+    override registerOpeningAnalyticsEvent() {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'Opt',
+        });
+    }
+
+    override registerCallbacks() {
+        this.eventHub.emit('requestSettings');
+        this.eventHub.emit('findCompilers');
+
+        this.container.on('shown', this.resize, this);
+
+        const cursorSelectionThrottledFunction = _.throttle(this.onDidChangeCursorSelection.bind(this), 500);
+        this.editor.onDidChangeCursorSelection(e => {
+            cursorSelectionThrottledFunction(e);
+        });
+    }
+
+    override onCompileResult(id: number, compiler, result) {
+        if (this.compilerInfo.compilerId !== id || !this.isCompilerSupported) return;
+        this.editor.setValue(result.source);
+        if (result.hasOptOutput) {
+            this.showOptResults(result.optOutput);
+        }
+        // TODO: This is unelegant again. Previously took advantage of fourth argument for the compileResult event.
```

Comment:
Note that the language is still passed in https://github.com/compiler-explorer/compiler-explorer/blob/9a75a7a1d686b3f5a59a20fa4426cc79e5bcb0bd/static/panes/compiler.js#L1139 so adding the missing argument to the base class _might_ work

Reply:
I'm going to assume this is OK to merge; as it was approved. Will give it a quick test in staging.